src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -126,6 +135,15 @@ public int HandleDatabaseRecordChange(DatabaseRecord record, long raftIndex)
                     _documentDatabase.RachisLogIndexNotifications.NotifyListenersAbout(raftIndex, e);
                     if (Logger.IsInfoEnabled)
                         Logger.Info($"Could not start index `{index.Name}`", e);
+
+                    try
+                    {
+                        index.Dispose();
```

Comment:
Why? You didn't managed to start the index (any reason) and you are disposing it

Reply:
The env is initialized even if we fail to start the index... so it would make sense to dispose it, no?